HSEARCH-4273 MassIndexer.cacheMode is not honored (#2701)
hibernate/hibernate-search: documentation/src/main/asciidoc/reference/mapper-orm-indexing-jsr352.asciidoc

The default is `IGNORE`, and it will be the most efficient choice in most cases,

> but using another mode such as `GET` may be more efficient if many of the entities being indexed
> refer to a small set of other entities.

Reviewer: Looks like this part was wrong:

```suggestion
but using another mode such as `GET` may be more efficient if many of the entities being indexed
are already present in the Hibernate ORM second-level cache before mass indexing.
```

> Enabling caches has an effect only if the entity has some lazy fields and/or the entity id is the document id too.

Reviewer: I don't think lazy fields have anything to do with this?

```suggestion
Enabling caches has an effect only if the entity id is also the document id, which is the default.
```

Author: Actually, I think that there could be some benefit, in the case the entity id was not the document id, but the entity had some lazy fields. But let's not confuse the user with that ;)

> `PUT` or `Normal` values may lead to a bad performance, because all the entities are also loaded into

Reviewer: ```suggestion
`PUT` or `NORMAL` values may lead to bad performance, because all the entities are also loaded into
```
